### ISSUE TITLE
Recursive functions never terminate in interpreter

### ISSUE THREAD

**Issue by jasonsbarr** [jasonsbarr/arith#9]
For some reason recursive functions just blow through the base case and continue recursing until exceeding the maximum stack size.

The base case is evaluated, but the function's execution doesn't terminate.

**Comment by jasonsbarr** [comment]
Example:

```racket
(define multiply
  (lambda (x y)
    (if (= y 1) x
      (+ x (multiply x (- y 1))))))
(multiply 5 3)
```

blows right through the base case in the interpreter. I put a `console.log` in the `makeLambda` function to see the actual args for each execution and saw something to the effect of

```javascript
[5, 3]
[5, 2]
[5, 1]
[5, 0]
[5, -1]
// ...
// a few hundred more calls
// Error: max call stack size exceeded
```

If I replace `x` in the base case with a `print` call I get the `print` call when the base case condition is met, but it doesn't return from the function - it just keeps decrementing `y` and doing recursive calls until the stack goes kaboom.

**Comment by jasonsbarr** [comment]
The plot thickens: transpiled recursive functions also show infinite recursion, so perhaps the problem is in the function I've defined to mimic `if` expressions.

**Comment by jasonsbarr** [comment]
No, the problem _is_ that I use a function to mimic `if` expressions. The stack keeps winding up even past the base case because that function call doesn't return. The base case evaluates, but the function doesn't return.

This is going to be fun to fix.

**Comment by jasonsbarr** [comment]
Fixed in #10
